### PR TITLE
Document the dependencies on linux

### LINKED ISSUE #46: Readme doesn't mention xkb dependency
Right now when I compile glazier, I get this error:

```sh
Package xkbcommon-x11 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `xkbcommon-x11.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'xkbcommon-x11' found
  ', build.rs:20:36
```

I can roughly guess what I need to do, but I feel like you shouldn't get a build error because of a missing dependency if the README doesn't mention you have to install that dependency.

### PR DESCRIPTION
**Pull request by jneem**
I've left out wayland and gtk dependencies, because those backends aren't working anyway.

Fixes #46

### CODE REVIEW

**Review by longmathemagician — APPROVED** [review]
It might be worth it to copy over the relevant section of the xilem readme (i.e. to provide common package names for the required libraries). Either way though this looks good to me.

**Comment by jneem** [comment]
Sure, makes sense. I made my best guess at the packages that need to be installed on fedora and debian/ubuntu. I'm not able to easily test the instructions, unfortunately...

File: README.md [hunk 1]
```diff
@@ -102,6 +102,27 @@ menus and support for IME.
...
+
+Most distributions already have `pkg-config` installed. To install the other packages
+on Fedora, run
+```
+sudo dnf install clang.x86_64 libxkbcommon-x11-devel.x86_64 libxcb-devel.x86_64 vulkan-loader.x86_64
+# This assumes you're using the x86_64 architecture; otherwise, append the appropriate value
+```
+To install the other packages on Debian or Ubuntu, run
+```
+sudo apt-get install clang libxkbcommon-x11-dev libxcb libvulkan-dev
```
Comment (longmathemagician) [inline README.md:123]: s/libxcb/pkg-config/

File: README.md [hunk 2]
```diff
@@ -102,6 +102,27 @@ menus and support for IME.
...
+work.
+
+On Linux and BSD, Glazier also requires `pkg-config` and `clang`,
+and the development packages of `libxkbcommon` and `libxcb`, to be installed.
+Some of the examples require `vulkan-loader`.
+
+Most distributions already have `pkg-config` installed. To install the other packages
+on Fedora, run
+```
+sudo dnf install clang.x86_64 libxkbcommon-x11-devel.x86_64 libxcb-devel.x86_64 vulkan-loader.x86_64
```
Comment (longmathemagician) [inline README.md:118]: sudo dnf install clang libxkbcommon-x11-devel libxcb-devel pkg-config vulkan-loader-devel

File: README.md (hunk 1 above)
Comment (longmathemagician) [reply README.md:121]: Great question, not actually sure about libxcb*. Testing with Ubuntu 22.10, I assume it got pulled in as a dep for libxkbcommon-x11-dev. At least on that version of ubuntu pkg-config is not installed by default.

I should probably also note that I was unable to actually run the examples (good ol' VM graphics adapter was not up to the task) but it did build.

**Pull request merged**